src/minichain/serialization.py

```
@@ -0,0 +1,66 @@
+"""Deterministic serialization helpers for consensus-critical data."""
+
+from __future__ import annotations
+
+import json
+from typing import Any, Mapping
```

Ruff (0.15.1)

[warning] Import from `collections.abc` instead: `Mapping`

Import from `collections.abc`

(UP035)